src/TortoiseProc/GitLogListBase.cpp

```diff
@@ -95,8 +95,16 @@ CGitLogListBase::CGitLogListBase():CHintListCtrl()
 	m_bFilterWithRegex = !!CRegDWORD(_T("Software\\TortoiseGit\\UseRegexFilter"), FALSE);
 	m_bFilterCaseSensitively = !!CRegDWORD(_T("Software\\TortoiseGit\\FilterCaseSensitively"), FALSE);
 
-	m_From = -1;
-	m_To = -1;
+	m_Filter.m_NumberOfLogsScale = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\LogDialog\\NumberOfLogsScale"), -1);
+	m_Filter.m_NumberOfLogs = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\LogDialog\\NumberOfLogs"), 0);

```

Comment:
About your `i1003-test` branch:
* I like the settings dialog
* Why is the From date surrounded with a border when no limitation is set?
* Do we need all those options in the "Last x .." entries in the context menu?
* Maybe add a checkbox entry in the context menu for "remember selected date on close"
* It does not compile ;)